File: app/src/main/java/com/meetferrytan/mvpdaggerstore/util/interceptors/MyResponseInterceptor.java
package com.meetferrytan.mvpdaggerstore.util.interceptors;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;


/**
 * Created by ferrytan on 11/3/16.
 */

// TODO implement your own Interceptor, depends on your API response
public class MyResponseInterceptor implements Interceptor {
    public MyResponseInterceptor() {
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        Response response = chain.proceed(request);

        return response;
    }
}

File: app/src/main/java/com/meetferrytan/mvpdaggerstore/util/interceptors/MyResponseInterceptor.kt
package com.meetferrytan.mvpdaggerstore.util.interceptors

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException


/**
 * Created by ferrytan on 11/3/16.
 */

// TODO implement your own Interceptor, depends on your API response
class MyResponseInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()

        return chain.proceed(request)
    }
}